plugins/module_utils/module_helper.py

```diff
@@ -133,10 +152,60 @@ def text(self):
         return self.msg or str(self.exc_val)
 
 
+class NamedDeprecation(object):
+    def __init__(self, deprecation):
+        self.name = None
+        self.deprecation = deprecation
+        self._prepared_deprecation = None
+        self.acked = False
+        self.triggered = False
+        self.module = None
+        self.ack_param = None
+
+    def prepare(self, name, module, ack_param):
+        self.name = name
+        self.module = module
+
+        d = list(self.deprecation)
+        d[0] = (
+            "{0}"
+            "\n\n"
+            "To prevent this message, add the deprecation name '{2}' to the '{1}' module parameter:\n"
+            "  {1}:\n"
+            "    - {2}"
+        ).format(d[0], ack_param, name)
+        self._prepared_deprecation = d
+
+    def acknowledge(self):
+        self.acked = True
+
+    def trigger(self):
+        if self.triggered or self.acked:
+            return
+
+        self.module.deprecate(*self._prepared_deprecation)  # pylint: disable=ansible-deprecated-no-version
```

Comment:
Please don't use args, but keyword args. If the signature of `AnsibleModule.deprecate` ever changes, this version would be screwed.

Reply:
Fixed here (and other places before we get to this point of the code), coming up in the next push. However, the signature of `AnsibleModule.deprecate()` governs the signature of `MH.PrepareNamedDeprecation.deprecate()`, so if it changes we're going to be screwed there in any case. Granted, it will be easier to handle different versions and signatures there, but it will require adjustments nonetheless.